rad/chains/issues.rad

```diff
@@ -0,0 +1,52 @@
+;; Helpers for setting up and interacting with issues chain (as defined in
+;; rad/issues-chain.rad)
+
+;; Create a remote issue chain. Returns a ref for the chain that you
+;; can assign to a variable.
+
+(def create-issues-chain!
+  "Create a remote issue chain with the given url. Returns a ref with the chain."
+  (fn [chain-name]
+    (send-prelude! chain-name)
+    (send-code! chain-name "rad/chains/issue-remote.rad")
+    (ref (new-chain chain-name))))
+
+(def new-issue!
+  "Create a new remote issue. Takes a chain ref, a keypair, an issue title and an issue body."
+  (fn [chain-ref keys title body]
+   (def chain (read-ref chain-ref))
+   (def id (uuid!))
+   (def msg
+        (string-append (lookup :url chain)
+                       id
+                       title
+                       body))
+   (def arg
+        {:id        id
+         :author    (lookup :public-key keys)
+         :title     title
+         :body      body
+         :signature (gen-signature! (lookup :private-key keys) msg)})
+   (send! (lookup :url chain)
+          [(list 'create-issue arg)])))
+
+(def list-issues
+  "Given an issues ref `x`, returns its issues. Does not itself update the chain."
+  (fn [chain-ref]
+    (lookup :result (eval-in-chain '(list-issues) (read-ref chain-ref)))))
```

Comment:
So something like:

```
(def query-chain
   (fn [chain expr]
      (lookup :result (eval-in-chain expr) (read-ref chain))))
```
?

(This means that instead of just `(list-issues chain)` we'd have to write `(query-chain chain '(list-issues))`, which is a little more annoying but not much.)

Reply:
:+1: